### PR TITLE
Enable nullability in some Event Args

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in some Event Args.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/4235)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/HtmlElementErrorEventArgs.cs [hunk 1]
```diff
@@ -40,6 +38,6 @@ internal HtmlElementErrorEventArgs(string description, string urlString, int lin
         /// <summary>
         ///  Url where error occurred
         /// </summary>
-        public Uri Url => _url ?? (_url = new Uri(_urlString));
+        public Uri? Url => _url ?? (_url = new Uri(_urlString));
```
Comment (weltkante) [inline src/System.Windows.Forms/src/System/Windows/Forms/HtmlElementErrorEventArgs.cs:41]: Is there a reason to make the API nullable? the implementation never will return null (it'll construct a new Uri)

Also there is new Syntax to support this lazy construction, but maybe keep it out of a nullability PR.

File: src/System.Windows.Forms/src/System/Windows/Forms/Control.ControlAccessibleObject.cs [hunk 2]
```diff
@@ -401,7 +401,7 @@ public override int GetHelpTopic(out string? fileName)
 
                 try
                 {
-                    topic = int.Parse(args.HelpKeyword, CultureInfo.InvariantCulture);
+                    topic = int.Parse(args.HelpKeyword!, CultureInfo.InvariantCulture);
                 }
                 catch (Exception e) when (!ClientUtils.IsCriticalException(e))
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/Control.ControlAccessibleObject.cs:406]: Replace with `int.TryParse` to simplify the code and avoid `try..catch`?

File: src/System.Windows.Forms/src/System/Windows/Forms/HtmlElementErrorEventArgs.cs [hunk 3]
```diff
@@ -12,9 +10,9 @@ namespace System.Windows.Forms
     public sealed class HtmlElementErrorEventArgs : EventArgs
     {
         private readonly string _urlString;
-        private Uri _url;
+        private Uri? _url;
 
-        internal HtmlElementErrorEventArgs(string description, string urlString, int lineNumber)
+        internal HtmlElementErrorEventArgs(string? description, string urlString, int lineNumber)
```
[the remark this answers is not in the document]
Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/HtmlElementErrorEventArgs.cs:15]: We use this in the Uri constructor which is not accepting nullable. Should I still change it?

[the remark this answers is not in the document]
Comment (RussKie) [reply src/System.Windows.Forms/src/System/Windows/Forms/HtmlElementErrorEventArgs.cs:15]: I had a look through the code, and there's very little information for `HTMLWindowEvents2` interface to make a good call.

Given that it worked like this for the past so many years I propose we can annotate the .ctor (and the rest of the type) as follows:
```suggestion
        internal HtmlElementErrorEventArgs(string description, string urlString, int lineNumber)
```

This way we retain the existing behaviour.

File: src/System.Windows.Forms/src/System/Windows/Forms/MeasureItemEventArgs.cs [hunk 4]
```diff
@@ -14,7 +12,8 @@ namespace System.Windows.Forms
     /// </summary>
     public class MeasureItemEventArgs : EventArgs
     {
-        public MeasureItemEventArgs(Graphics graphics, int index) : this(graphics, index, 0)
+        public MeasureItemEventArgs(Graphics graphics, int index)
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/MeasureItemEventArgs.cs:15]: Should this be?
```suggestion
        public MeasureItemEventArgs(Graphics? graphics, int index)
```

**Review by RussKie — APPROVED** [review]

**Comment by RussKie** [comment]
Thank you!

**Pull request merged**